c10/core/SymInt.h

```diff
@@ -47,12 +48,46 @@ class C10_API SymInt {
     return data_ == p2.data_;
   }
 
+  bool operator!=(const SymInt& p2) const {
+    return data_ != p2.data_;
+  }
+
   SymInt operator+(SymInt sci) const {
     TORCH_CHECK(
         !this->is_symbolic() && !sci.is_symbolic(),
         "Symbolic Add isn't supported yet");
     return SymInt(data_ + sci.data_);
   }
+  bool operator<(SymInt sci) const {
+    TORCH_CHECK(
+        !this->is_symbolic() && !sci.is_symbolic(),
+        "Symbolic lt isn't supported yet");
+    return data_ < sci.data_;
+  }
+  void operator*=(SymInt sci) {
+    TORCH_CHECK(
+        !this->is_symbolic() && !sci.is_symbolic(),
+        "Symbolic mul_ isn't supported yet");
+    data_ = data_ * sci.data_;
+  }
+
+  bool operator<(int64_t sci) const {
+    TORCH_CHECK(!this->is_symbolic(), "Symbolic lt isn't supported yet");
+    return data_ < sci;
+  }
+  bool operator==(int64_t sci) const {
+    TORCH_CHECK(!this->is_symbolic(), "Symbolic eq isn't supported yet");
+    return data_ == sci;
+  }
+  bool operator!=(int64_t sci) const {
+    TORCH_CHECK(!this->is_symbolic(), "Symbolic neq isn't supported yet");
+    return data_ != sci;
+  }
+
+  SymInt operator*(int64_t sci) const {
+    TORCH_CHECK(!this->is_symbolic(), "Symbolic mul isn't supported yet");
```

Comment:
We are going to want this out-of-line in a cpp file so we aren't inlining a big blob of exception throwing code everywhere.